src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -324,6 +324,11 @@ private ImmutableArray<ISymbol> GetSymbolsOffOfName(NameSyntax name)
         private ImmutableArray<ISymbol> GetSymbolsOffOfExpression(ExpressionSyntax originalExpression)
         {
             var expression = originalExpression.WalkDownParentheses();
+            if (expression is AwaitExpressionSyntax awaitExpression)
```

Comment:
i'm not exactly understanding why this is the right fix.  can you provide insight as to what's actually going wrong when you have `await x..`?

Reply:
I'm guessing it's the order of operations. Prior to adding something between the dots, you have a parse tree that is effectively `(await x)..z`, but you are asking for completion of an expression that would change the parse tree to `await (x.y.z)`.

Reply:
i see.  in that case we should explicitly check for this *and only this* tree and do the fixing up accordingly.

Reply:
Thank you! 
We are fixing here (await x).$$.z not (await x)..$$z. Please let me know if I misunderstood the context of "this and only this tree".

---
In reply to: [335224801](https://github.com/dotnet/roslyn/pull/39302#discussion_r335224801) [](ancestors = 335224801)

Reply:
What I don't get though is why is this fixing it? I would have thought we had to do some work here to reconstruct a different tree somehow -- the compiler should already be walking through parenthesis accordingly.

Reply:
Maybe it's just because in this scenario it's not clear to me which expression "originalExpression" is.

Reply:
> What I don't get though is why is this fixing it? I would have thought we had to do some work here to reconstruct a different tree somehow -- the compiler should already be walking through parenthesis accordingly.

`originalExpression` is `await request` which could be surrounded by `()` as well. We need request to run 
`_context.SemanticModel.GetSymbolInfo(expression, _cancellationToken);`
for it.
Calling `GetSymbolInfo` for `await request` does not bring any symbols or candidates.

Reply:
> Calling GetSymbolInfo for await request does not bring any symbols or candidates.

That makes sense to me.  This shoudl be fallign back to the codepath where we call GetTypeInfo and operate on the result of that.  Something still seems fishy to me here.

Reply:
@CyrusNajmabadi Chatting with Ivan that my same (independent) thought. It looks like that's not the case though: it looks like GetSymbolsOffOfBoundExpression assumes the symbol info was _something_, and if it's not then it just prematurely bails. That seems like a bug. I get that it's trying to do special cases where something might have a type that isn't a "useful" answer to return, but it shouldn't be doing that unilaterally. I don't know if this was a situation where we felt the compiler too often gave us GetTypeInfos for things that we didn't otherwise didn't make sense that we figured we should always get another symbol....

Reply:
The flow seems odd in other ways too: it seems odd that we'd call both GetSymbolInfo, get a symbol that we know will cause us to ignore the result of GetTypeInfo, and then we call GetTypeInfo anyways...

Reply:
I'm also not sure if we need tests too around conditional access or (ugh) pointer dereferencing. It seems like those all will have the same bug since they're all calling into GetSymbolsOffOfBoundExpression?

Reply:
Oh now I'm confused looking at the code too:

https://github.com/dotnet/roslyn/pull/39302/files#diff-5c40ed40c174e912b51b8c61d93e6ef6R473-R480

So yes, if we have no symbol, then the type info ("container" in that code) should be getting used. So either we got no symbol nor type info, or something else is up.

Reply:
But it's very common to have no symbol for an expression.  Basically anything that generates temp expressions can cause this.  So we always need to try the 'type info' path.  Now, why we have both a symbol-info-path and type-info path is not something i recall anymre.